hastory-data-gen/src/Data/Hastory/Gen.hs

```diff
@@ -14,6 +17,26 @@ instance Arbitrary Entry where
   arbitrary = genValid
   shrink = shrinkValidStructurally
 
+instance GenValid Password where
+  genValid = mkPassword <$> genValid
+  shrinkValid _ = [] -- don't shrink Password
+
 instance GenValid SyncRequest where
   genValid = genValidStructurally
   shrinkValid = shrinkValidStructurally
+
+instance GenValid UserForm where
+  genValid = genValidStructurally
+  shrinkValid = shrinkValidStructurally
+
+instance GenValid Username where
+  genValid = Username <$> userNameTextGen
+    where
+      userNameTextGen = oneof [lengthFourText, arbitraryLengthText]
```

Comment:
@NorfairKing Should the definition of `userNameTextGen = oneof [lengthFourText, arbitraryLengthText]` instead just be `userNameTextGen = arbitraryLengthText`?

Reply:
No, see what I just commented above. The length must be at least 3 so that wouldn't work.